src/protocols/screencopy.rs

```diff
@@ -42,7 +47,22 @@ impl ScreencopyManagerState {
         };
         display.create_global::<D, ZwlrScreencopyManagerV1, _>(VERSION, global_data);
 
-        Self
+        Self {
+            damage_trackers: HashMap::new(),
+        }
+    }
+
+    pub fn damage_tracker(&mut self, output: &Output) -> Option<&mut ScreencopyDamageTracker> {
+        self.damage_trackers.get_mut(output)
+    }
+
+    pub fn damage_tracker_or_insert(&mut self, output: &Output) -> &mut ScreencopyDamageTracker {
+        self.damage_trackers
+            .entry(output.clone())
+            .or_insert_with(|| ScreencopyDamageTracker {
+                damage_tracker: OutputDamageTracker::from_output(output),
```

Comment:
I guess if a region copy is requested then this should be a small region-sized damage tracker, rather than a full output sized damage tracker? Or alternatively you can clip the damage when rendering I guess with `Rectangle::intersect`.